### PR TITLE
feat: Add cozy-authentication

### PR DESCRIPTION
**Pull request by ptbrowne**
Extract cozy-authentication from Drive.

- Locales are extracted from Drive (an improvement would be to pull them from Transifex) through a manually launched script
- Cozy-authentication is now transpiled instead of bundled. This avoids the use of webpack's externals that we used before.

I don't understand why the tests are failing on travis, this is why I put "test": "true" and added unittest. If you can confirm with me that it's working on your machines, I would be grateful :)

### CODE REVIEW

**Comment by y-lohse** [comment]
I have the same errors on my machine than on travis (https://travis-ci.org/cozy/cozy-libs/builds/506731734#L986). Trying to find what's going on.

File: packages/cozy-authentication/src/steps/ButtonLinkRegistration.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import { getPlatform } from 'cozy-device-helper'
+import { Button } from 'cozy-ui/transpiled/react'
+import flag from 'cozy-flags'
```
Comment (y-lohse) [inline packages/cozy-authentication/src/steps/ButtonLinkRegistration.jsx]: We need to add `cozy-flags` as a dependency.

**Comment by y-lohse** [comment]
I managed to run the tests on my machine… but it's weird?

The dependencies used in cozy-authentication that are also part of this monorepo (`cozy-device-helper` and `cozy-flags`) seemed to be installed with symbolic links, so they didn't have their `dist/` files. After I ran `yarn build` in both these projects, the tests in `cozy-authentication` passed.

So I'm guessing on your machine, you had already built the other packages, and that's why it ran. However `cozy-harvest` is in the same situation with `cozy-realtime` but somehow this works..?

**Comment by ptbrowne** [comment]
Ah so we should run yarn build before yarn test in travis.yml ? Thanks for finding this ! I will try.

File: packages/cozy-authentication/package.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    "@babel/core": "^7.3.4",
+    "babel-plugin-css-modules-transform": "^1.6.2",
+    "babel-preset-cozy-app": "^1.5.0",
+    "cozy-ui": "^19.12.0",
+    "date-fns": "^1.29.0",
+    "enzyme": "^3.9.0",
+    "enzyme-adapter-react-16": "^1.11.2",
+    "eslint-cli": "1.1.1",
+    "jest": "24.5.0",
+    "parcel": "^1.10.3",
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply packages/cozy-authentication/package.json]: Yes it is used for example page.

**Comment by y-lohse** [comment]
[App mobile onboarding - externaliser code lib externe](https://trello.com/c/j0RUO0aT/1719-app-mobile-onboarding-externaliser-code-lib-externe)

**Comment by ptbrowne** [comment]
Yes it worked ! Thanks !

**Pull request merged**